src/bp/core/services/dialog/flow/service.ts

```diff
@@ -209,6 +263,30 @@ export class FlowService {
     this.realtime.sendToSocket(payload)
   }
 
+  private async _testAndLockMutex(botId: string, currentFlowEditor: string, flowLocation: string): Promise<FlowMutex> {
+    const currentMutex = ((await this.kvs.get(botId, flowLocation)) || {}) as FlowMutex
+    let { lastModifiedBy: flowOwner, lastModifiedAt } = currentMutex
+
+    // TODO: might use somthing else than flowLocation as key. Maybe something like `FLOWMUTEX: ${flowLocation}`
+    if (currentFlowEditor === flowOwner) {
+      const mutex = { lastModifiedBy: flowOwner, lastModifiedAt: new Date() }
+      await this.kvs.set(botId, flowLocation, mutex)
+      return mutex
+    }
+
+    const isMutexExpired =
+      moment(lastModifiedAt)
+        .add(MUTEX_LOCK_DELAY_SECONDS, 'seconds')
+        .unix() < moment().unix()
```

Comment:
Duplicated logic with above (remainingSeconds).  If using a get, you could just check if remainingTime > 0 ?